substrate/frame/support/procedural/src/construct_runtime/expand/origin.rs

```diff
@@ -301,6 +305,22 @@ pub fn expand_outer_origin(
 			}
 		}
 
+		impl #scrate::__private::AsSystemOriginSigner<<#runtime as #system_path::Config>::AccountId> for RuntimeOrigin {
+			fn as_system_origin_signer(&self) -> Option<&<#runtime as #system_path::Config>::AccountId> {
+				if let OriginCaller::system(#system_path::Origin::<#runtime>::Signed(ref signed)) = &self.caller {
+					Some(signed)
+				} else {
+					None
+				}
+			}
+		}
+
+		impl #scrate::__private::AsAuthorizedOrigin for RuntimeOrigin {
+			fn is_authorized(&self) -> bool {
```

Comment:
I updated the naming and the docs with a more detailed explanation. This is the state of things:

Extrinsics need to be validated and authorized before making it on chain. Today there are 2 types of extrinsics:
- inherents - validated through `ProvideInherent` and the supporting code in `construct_runtime!`
- transactions - validated as follows
    - through `ValidateUnsigned` for traditionally unsigned transactions
    - through `TransactionExtensions` for traditionally signed and general transactions

All inherents are bare and do not do any origin authorization. They are valid and authorized if their call is an inherent (`ProvideInherent::is_inherent` returns `true`) and it passes the checks in `ProvideInherent::check_inherent`.

Unsigned transactions today are validated through `ValidateUnsigned`, but in the future they will be validated through `TransactionExtension`s as general transactions and will have to have their origin authorized, with that origin no longer being an account, but a source within the system, like a pallet. A solution for this is detailed in https://github.com/paritytech/polkadot-sdk/issues/2415 .

Traditionally signed transactions have their origin authorized by the `Checkable::check` impl of `UncheckedExtrinsic`, which validates the provided signature and dispatches with the signer's origin. This logic will also be moved to the general transaction type, where we will have a specific extension doing this signature verification work.

General transactions rely on their `TransactionExtension` pipeline to authorize the origin of the transaction. The rule here is that after all the extensions ran their validation logic, there are no more origin mutations possible, so the final origin must have been authorized by some extension, otherwise the transaction is invalid (similar to how you would try to submit a `Balances::transfer_allow_death` as an unsigned transaction).

This transition period between this PR and the deprecation of `ValidateUnsigned`, which is phase 2 of Extrinsic Horizon, will come with some inconsistent naming of things, but it can't be done in one go, this PR is already way to big as it is. We already have a WIP for phase 2 and a clear direction there, so it should solve all of these problems.